extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -48,19 +49,36 @@
   private static final Logger LOG = new Logger(OpenCensusProtobufInputRowParser.class);
 
   private static final String SEPARATOR = "-";
-  public static final String NAME = "name";
-  public static final String VALUE = "value";
-  public static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String NAME = "name";
+  private static final String VALUE = "value";
+  private static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String DEFAULT_RESOURCE_PREFIX = "resource.";
   private final ParseSpec parseSpec;
   private final List<String> dimensions;
 
-  @JsonCreator
+  private final String metricDimension;
+  private final String metricLabelPrefix;
+  private final String resourceLabelPrefix;
+  private final boolean hasMetricDimension;
+  private final boolean hasMetricLabelPrefix;
+  private final boolean hasResourceLabelPrefix;
+
   public OpenCensusProtobufInputRowParser(
-      @JsonProperty("parseSpec") ParseSpec parseSpec
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("metricDimension") String metricDimension,
+      @JsonProperty("descriptorDimensionPrefix") String metricPrefix,
+      @JsonProperty("resourceDimensionPrefix") String resourcePrefix
   )
   {
     this.parseSpec = parseSpec;
     this.dimensions = parseSpec.getDimensionsSpec().getDimensionNames();
+    this.metricDimension = metricDimension;
+    this.metricLabelPrefix = metricPrefix;
+    this.resourceLabelPrefix = resourcePrefix == null ? DEFAULT_RESOURCE_PREFIX : resourcePrefix;
+
+    this.hasMetricDimension = !Strings.isNullOrEmpty(this.metricDimension);
+    this.hasMetricLabelPrefix = !Strings.isNullOrEmpty(this.metricLabelPrefix);
```

Comment:
I don't think we need those variables, you can achieve the same behavior by setting `this.metricLabelPrefix = Strings.nullToEmpty(metricLabelPrefix)`

Reply:
Done.